activerecord/test/cases/autosave_association_test.rb

```diff
@@ -517,6 +517,24 @@ def test_invalid_adding_before_save
     assert_not_predicate new_firm, :persisted?
   end
 
+  def test_adding_unsavable_association
+    new_firm = Firm.new("name" => "A New Firm, Inc")
+    client = new_firm.clients.new("name" => "Apple")
+    # Stub the save method of the client to simulate aborting the insert
```

Comment:
Is this simulating `throw :abort` in a before callback? Could we make that happen for real? Rails' tests are generally written end-to-end, using only the public API.

Reply:
I actually came across the bug addressed in this PR trying to fix https://github.com/Shopify/activerecord-rescue_from_duplicate/issues/13, and I just replicated it's [behaviour](https://github.com/Shopify/activerecord-rescue_from_duplicate/blob/ca140ad0102db088ba671573d9490e96d92c344c/lib/rescue_from_duplicate/active_record/extension.rb#L29), but I think it will work with `throw :abort` in a `before_save` just as well.